File: kata-solutions/tell-dont-ask/tell-dont-ask-python/src/useCase/OrderApprovalUseCase.py
from src.repository.OrderRepository import OrderRepository
from src.useCase.OrderApprovalRequest import OrderApprovalRequest

class OrderApprovalUseCase:
    def __init__(self, order_repository: OrderRepository):
        self.order_repository = order_repository

    def run(self, request: OrderApprovalRequest):
        order = self.order_repository.get_by_id(request.get_order_id())

        if request.is_approved():
          order.approve()
        else:
          order.reject()

        self.order_repository.save(order)

File: kata-solutions/tell-dont-ask/tell-dont-ask-java/app/src/main/java/useCase/OrderApprovalUseCase.java
package useCase;

import domain.Order;
import repository.OrderRepository;

public class OrderApprovalUseCase {
    private final OrderRepository orderRepository;

    public OrderApprovalUseCase(OrderRepository orderRepository) {
        this.orderRepository = orderRepository;
    }

    public void run(OrderApprovalRequest request) {
        final Order order = orderRepository.getById(request.getOrderId());

        if (request.isApproved())
            order.approve();
        else
            order.reject();

        orderRepository.save(order);
    }

}
